Sources/Data Model/OptimizelyJSON.swift

```diff
@@ -0,0 +1,144 @@
+* distributed under the License is distributed on an "AS IS" BASIS,        *
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+* See the License for the specific language governing permissions and      *
+* limitations under the License.                                           *
+***************************************************************************/
+
+import Foundation
+
+class OptimizelyJSON {
+    
+    private typealias schemaHandler = (Any) throws -> Void
+    private var payload: String?
+    private var data = [String: Any]()
+    
+    // MARK: - Init
+    
+    init(payload: String) throws {
+        do {
+            guard let data = payload.data(using: .utf8),
+                let jsonData = try JSONSerialization.jsonObject(with: data, options : .allowFragments) as? [String: Any] else
+            {
+                throw OptimizelyError.failedToConvertStringToDictionary
+            }
+            self.data = jsonData
+            self.payload = payload
+        } catch {
+            throw OptimizelyError.failedToConvertStringToDictionary
+        }
+    }
+    
+    init(data: [String: Any]) throws {
+        if !JSONSerialization.isValidJSONObject(data) {
+            throw OptimizelyError.invalidDictionary
+        }
+        self.data = data
+    }
+    
+    private init() {}
+    
+    // MARK: - OptimizelyJSON Implementation
+    
+    /// - Returns:  the string representation of json
+    /// - Throws: `OptimizelyError`
+    func toString() throws -> String {
+        guard let payload = self.payload else {
+            do {
+                let jsonData = try JSONSerialization.data(withJSONObject: data, options: [])
+                if let jsonString = String(data: jsonData, encoding: .utf8) {
+                    self.payload = jsonString
+                    return jsonString
+                }
+            } catch {
+            }
+            throw OptimizelyError.failedToConvertDictionaryToString
+        }
+        return payload
+    }
+    
+    /// - Returns:  the json dictionary
+    /// - Throws: `OptimizelyError`
+    func toMap() throws -> [String: Any] {
+        return data
+    }
+    
+    /// populates the decodable schema passed by the user
+    /// - Parameters:
+    ///   - jsonPath: Key path for the value.
+    ///   - schema: Decodable schema to populate.
+    /// - Throws: `OptimizelyError`
+    func getValue<T: Decodable>(jsonPath: String, schema: inout T) throws {
+        func populateDecodableSchema(value: Any) throws {
+            // Try to decode value into schema
+            if JSONSerialization.isValidJSONObject(value) {
+                do {
+                    let jsonData = try JSONSerialization.data(
+                        withJSONObject: value,
+                        options: .prettyPrinted
+                    )
+                    schema = try JSONDecoder().decode(T.self, from: jsonData)
+                    return
+                } catch {
```

Comment:
Why we are swallowing error?

Reply:
Since we try to assign value directly to the generic operator if the above case fails, if both fail, then we throw a generalised error here:  
https://github.com/optimizely/swift-sdk/pull/315/files/b4bd90cd23dbc74c17edef1c776244e80ad55b16#diff-fbd9b0e0f2ff68ec3188bdee23588907R99